### PR TITLE
Step 4 (SSR)

### PR DESCRIPTION
**Pull request by IggyPope**
# Features

- Implemented SSR for the Home page
- Implemented SSR for the Article page

## Checklist before requesting a review

- [x] I have tested changes locally;
- [x] I have followed styleguide;
- [x] I have performed a self-review of my code.

### CODE REVIEW

File: src/api/news/queries.ts [hunk 1]
```diff
@@ -2,18 +2,20 @@ import { useQuery } from '@tanstack/react-query';
 import { Sections } from '@/constants/routes';
 import { fetchArticleByUri, fetchArticles } from './module';
 
-enum QueryKeys {
+export enum QueryKeys {
```
Comment (Ilya-Valasiuk) [inline src/api/news/queries.ts]: do not export this enum.

Create method here like `prefetchArticles`

File: src/pages/article/index.tsx [hunk 2]
```diff
@@ -1,19 +1,47 @@
...
-}: InferGetServerSidePropsType<typeof getServerSideProps>) {
   if (!(uri && typeof uri === 'string')) {
-    return <div>Article not found</div>;
+    return {
+      notFound: true,
+    };
   }
 
-  return <Article uri={uri}></Article>;
+  const queryClient = new QueryClient();
```
Comment (Ilya-Valasiuk) [inline src/pages/article/index.tsx]: move queryClient to `src/constants/query-client.ts`

File: src/pages/article/index.tsx [hunk 3]
```diff
@@ -1,19 +1,47 @@
...
+}) satisfies GetServerSideProps<{
+  uri: string;
+  dehydratedState: DehydratedState;
+}>;
+
+type Props = InferGetServerSidePropsType<typeof getServerSideProps>;
+
+export default function ArticlePage({ uri, dehydratedState }: Props) {
+  return (
+    <HydrationBoundary state={dehydratedState}>
```
Comment (Ilya-Valasiuk) [inline src/pages/article/index.tsx]: HydrationBoundary can be moved to upper level (_app.tsx)

**Review by Ilya-Valasiuk — CHANGES_REQUESTED** [review]

File: src/api/news/queries.ts (hunk 1 above)
Comment (IggyPope) [reply src/api/news/queries.ts]: Done. Created prefetchArticles() and prefetchSingleArticle() methods

File: src/pages/article/index.tsx (hunk 2 above)
Comment (IggyPope) [reply src/pages/article/index.tsx]: done

File: src/pages/article/index.tsx (hunk 3 above)
Comment (IggyPope) [reply src/pages/article/index.tsx]: done

**Pull request merged**